core/src/main/scala/akka/kafka/ProducerMessage.scala

```diff
@@ -35,8 +35,10 @@ object ProducerMessage {
    * `offset` of the produced message.
    */
   final case class Result[K, V, PassThrough](
-    offset: Long,
-    message: Message[K, V, PassThrough]
-  )
+      metadata: RecordMetadata,
+      message: Message[K, V, PassThrough]
+  ) {
+    def offset = metadata.offset()
```

Comment:
Should we maybe be selective with what we allow access to rather than provide the full `Metadata` to the consumer code (promising we always have an instance of the actual kafka client `Metadata` type may be limiting with regards to evolving the API)?